src/sentry/workflow_engine/handlers/detector/stateful.py

```diff
@@ -156,12 +183,14 @@ def evaluate(
             )
             if result:
                 results[result.group_key] = result
+
+        self.commit_state_updates()
```

Comment:
just for my understanding, what happens if we fail when committing state updates?

Reply:
i guess it would affect subsequent evals 😨